test/tests/VehicleTest.cpp

```diff
@@ -0,0 +1,1218 @@
+            vehicle->mini_golf_current_animation = 0;
+            vehicle->mini_golf_flags = 0;
+        }
+        if (vehicleEntry->flags & VEHICLE_ENTRY_FLAG_4)
+        {
+            if (vehicle->IsHead())
+            {
+                subposition = VehicleTrackSubposition::ReverserRCFrontBogie;
+            }
+        }
+        if (vehicleEntry->flags & VEHICLE_ENTRY_FLAG_5)
+        {
+            subposition = VehicleTrackSubposition::ReverserRCRearBogie;
+        }
+        vehicle->TrackSubposition = subposition;
+
+        auto chosenLoc = carPosition;
+        vehicle->TrackLocation = chosenLoc;
+
+        int32_t direction = trackElement->GetDirection();
+        vehicle->sprite_direction = direction << 3;
+
+        if (ride->type == RIDE_TYPE_SPACE_RINGS)
+        {
+            direction = 4;
+        }
+        else
+        {
+            if (ride_type_has_flag(ride->type, RIDE_TYPE_FLAG_VEHICLE_IS_INTEGRAL))
+            {
+                if (RideTypeDescriptors[ride->type].StartTrackPiece != FLAT_TRACK_ELEM_1_X_4_B)
+                {
+                    if (RideTypeDescriptors[ride->type].StartTrackPiece != FLAT_TRACK_ELEM_1_X_4_A)
+                    {
+                        if (ride->type == RIDE_TYPE_ENTERPRISE)
+                        {
+                            direction += 5;
+                        }
+                        else
+                        {
+                            direction = 4;
+                        }
+                    }
+                }
+            }
+        }
+
+        chosenLoc += CoordsXYZ{ word_9A2A60[direction], RideTypeDescriptors[ride->type].Heights.VehicleZOffset };
+
+        vehicle->current_station = trackElement->GetStationIndex();
+
+        vehicle->MoveTo(chosenLoc);
+        vehicle->SetTrackType(trackElement->GetTrackType());
+        vehicle->SetTrackDirection(vehicle->sprite_direction >> 3);
+        vehicle->track_progress = 31;
+        if (vehicleEntry->flags & VEHICLE_ENTRY_FLAG_MINI_GOLF)
+        {
+            vehicle->track_progress = 15;
+        }
+        vehicle->update_flags = VEHICLE_UPDATE_FLAG_COLLISION_DISABLED;
+        if (vehicleEntry->flags & VEHICLE_ENTRY_FLAG_HAS_INVERTED_SPRITE_SET)
+        {
+            if (trackElement->IsInverted())
+            {
+                vehicle->SetUpdateFlag(VEHICLE_UPDATE_FLAG_USE_INVERTED_SPRITES);
+            }
+        }
+        vehicle->SetState(Vehicle::Status::MovingToEndOfStation);
+    }
+
+    // loc_6DDD5E:
+    vehicle->num_peeps = 0;
+    vehicle->next_free_seat = 0;
+    vehicle->BoatLocation.setNull();
+    return vehicle;
+}
+
+rct_ride_entry* Ride::GetRideEntry() const
+{
+    return get_ride_entry(subtype);
+}
+
+TEST(Vehicle, VehicleCreateSimple)
+{
+    reset_sprite_list();
+    Ride& r = _rideList[0];
+    r = Ride{};
+    r.id = 0;
+    r.type = RIDE_TYPE_MONORAIL;
+    int32_t rd{};
+    TrackElement te{};
+    vehicle_create_car(0, 0, 0, 0, {}, &rd, &te);
```

Comment:
There is some minimal configuration necessary to call this function, otherwise it could segfault